src/module.rs

```diff
@@ -38,54 +102,376 @@ impl_function_args!(A, B, C, D, E, F, G, H, I, J, K, L, M, N, O, P, Q, R, S);
 #[derive(Default)]
 pub struct TranslatedModule {
     translated_code_section: Option<TranslatedCodeSection>,
+    types: SimpleContext,
+    // TODO: Should we wrap this in a `Mutex` so that calling functions from multiple
+    //       threads doesn't cause data races?
+    table: Option<(TableType, Vec<u32>)>,
+    memory: Option<MemoryType>,
+}
+
+pub fn quickhash<H: Hash>(h: H) -> u64 {
+    let mut hasher = std::collections::hash_map::DefaultHasher::new();
+    h.hash(&mut hasher);
+    hasher.finish()
 }
 
 impl TranslatedModule {
-    // For testing only.
-    // TODO: Handle generic signatures.
-    pub unsafe fn execute_func<Args: FunctionArgs, T>(&self, func_idx: u32, args: Args) -> T {
+    pub fn instantiate(mut self) -> ExecutableModule {
+        let table = {
+            let code_section = self
+                .translated_code_section
+                .as_ref()
+                .expect("We don't currently support a table section without a code section");
+            let types = &self.types;
+
+            self.table
+                .as_mut()
+                .map(|&mut (_, ref mut idxs)| {
+                    let initial = idxs
+                        .iter()
+                        .map(|i| {
+                            let start = code_section.func_start(*i as _);
+                            let ty = types.func_type(*i);
+
+                            RuntimeFunc {
+                                func_start: start,
+                                sig_hash: quickhash(ty) as u32,
```

Comment:
Is it the case that using `quickcheck` here is just a temporary solution to get the code up and running?

Reply:
Yes! I want to transition to having Lightbeam support Wasmtime's method of signature checking but I haven't got around to it yet. This is just a stopgap that allows us to check that signature checking works in theory.